AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/mrecs/RecyclerViewMrecAdActivity.java

```diff
@@ -0,0 +1,243 @@
+package com.applovin.enterprise.apps.demoapp.ads.max.mrecs;
+
+import android.app.Activity;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+import android.widget.TextView;
+
+import com.applovin.enterprise.apps.demoapp.R;
+import com.applovin.mediation.MaxAd;
+import com.applovin.mediation.MaxAdFormat;
+import com.applovin.mediation.MaxAdViewAdListener;
+import com.applovin.mediation.MaxError;
+import com.applovin.mediation.ads.MaxAdView;
+import com.applovin.sdk.AppLovinSdkUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+public class RecyclerViewMrecAdActivity
+        extends AppCompatActivity
+        implements MaxAdViewAdListener
+{
+    private final int                   AD_VIEW_COUNT = 5;
+    private final int                   AD_INTERVAL   = 10;
+    private final ArrayList<String>     sampleData    = new ArrayList<>( Arrays.asList( "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".split( "" ) ) );
+    private final List<MaxAdView>       adViews       = new ArrayList<>( AD_VIEW_COUNT );
+    private       CustomRecyclerAdapter adapter;
+
+    @Override
+    protected void onCreate(@Nullable final Bundle savedInstanceState)
+    {
+        super.onCreate( savedInstanceState );
+        setContentView( R.layout.activity_mrec_recycler_view );
+        setTitle( R.string.activity_recycler_view_mrecs );
+
+        adapter = new CustomRecyclerAdapter( this, sampleData );
+
+        // Configure recycler view
+        RecyclerView recyclerView = findViewById( R.id.mrec_recycler_view );
+        recyclerView.setLayoutManager( new LinearLayoutManager( this ) );
+        recyclerView.setAdapter( adapter );
+
+        configureAdViews();
+    }
+
+    private void configureAdViews()
+    {
+        // Insert rows at each interval to be used to display an ad
+        for ( int i = 0; i < sampleData.size(); i += AD_INTERVAL )
+        {
+            sampleData.add( i, "" );
+            adapter.notifyItemInserted( i );
+        }
+
+        for ( int i = 0; i < AD_VIEW_COUNT; i++ )
+        {
+            MaxAdView adView = new MaxAdView( "YOUR_AD_UNIT_ID", MaxAdFormat.MREC, this );
+            adView.setListener( this );
+
+            // Set this extra parameter to work around SDK bug that ignores calls to stopAutoRefresh()
+            adView.setExtraParameter( "allow_pause_auto_refresh_immediately", "true" );
+            adView.stopAutoRefresh();
+
+            // Load the ad
+            adView.loadAd();
+            adViews.add( adView );
+        }
+    }
+
+    //region RecyclerAdapter and ViewHolder
```

Comment:
We do add newlines before and after these region comments though (check out the other activities for example)
```suggestion
    //region RecyclerAdapter and ViewHolder
    
```